File: go/p7/p7_v1.go
package main

import (
	"fmt"
	"time"
)

func problem7() int {
	var isPrime = func (number int, primer_numbers []int) bool {
		for _, prime := range primer_numbers {
			if number%prime == 0 {
				return false
			}
		}
		return true
	}

	const n int = 10001

	primeNumbers := []int{2}

	primeCandidate := 3

	for len(primeNumbers) < n {
		if isPrime(primeCandidate, primeNumbers) {
			primeNumbers = append(primeNumbers, primeCandidate)
		}
		primeCandidate += 2
	}

	return primeNumbers[n-1]
}

func main() {
	start := time.Now()
	fmt.Println("Result:", problem7())
	elapsed := time.Since(start)
	fmt.Printf("Execution time: %.6f seconds\n", elapsed.Seconds())
}

File: python/p7/p7_v1.py
def problem7():
    def is_prime(number, prime_numbers):
        for i in prime_numbers:
            if number % i == 0:
                return False
        return True

    n = 10001
    prime_numbers = [2]
    current_number = 3
    while len(prime_numbers) != n:
        if is_prime(current_number, prime_numbers):
            prime_numbers.append(current_number)
        current_number += 2

    return prime_numbers[n - 1]

if __name__ == "__main__":
    import time
    
    start_time = time.time()
    result = problem7()
    end_time = time.time()

    print(f"Result: {result}")
    print(f"Execution time: {end_time - start_time:.6f} seconds")